### ISSUE TITLE
Determine overhead of pyjnius

### ISSUE THREAD

**Issue by lintool** [castorini/anserini#604]
We should compare retrieval from native Java vs. retrieval using pyjnius from Python to find out the overhead.

**Comment by lintool** [comment]
We have to have a preliminary answer to this?
https://github.com/castorini/anserini/pull/644

**Comment by lintool** [comment]
Ref #692 - Fixed timing bug. The Java version is actually 2x slower, not 5x faster.
Have a hypothesis of why this is so... will follow up.

**Comment by lintool** [comment]
Interesting - these are results on `damiano`:

Java:

```
...
Retrieving query 6950 (0.087 s/query)
Retrieving query 6960 (0.087 s/query)
Retrieving query 6970 (0.087 s/query)
Done!
```

Python:

```
...
Retrieving query 6950 (0.081 s/query)
Retrieving query 6960 (0.081 s/query)
Retrieving query 6970 (0.081 s/query)
Done!
```

So, on Linux, the Java version is slightly slower.

The previous reports of Java being 2x slower than Python is on my iMac Pro.

Huh?

**Comment by lintool** [comment]
Ref #695 

Current implementation, running on my iMac Pro - Python version:

```
Retrieving query 6600 (0.056 s/query)
Retrieving query 6700 (0.056 s/query)
Retrieving query 6800 (0.056 s/query)
Retrieving query 6900 (0.056 s/query)
Done!
```
Java version:
```
Retrieving query 6600 (0.052 s/query)
Retrieving query 6700 (0.052 s/query)
Retrieving query 6800 (0.052 s/query)
Retrieving query 6900 (0.052 s/query)
Done!
```
These results make sense now: Python adds a small overhead.

**Comment by lintool** [comment]
tl;dr - pyjnius has a relatively small, but measurable impact on latency.

**Comment by lintool** [comment]
Update to Lucene 8:

Python: `Retrieving query 6900 (0.054 s/query)`
Java: `Retrieving query 6900 (0.050 s/query)`

Tiny bit faster...
